### PR TITLE
Refactor Graph

### PR DESCRIPTION
**Pull request by 3v0k4**
* extract graph from Angular
* make explicit different code paths for mobile / fixed session drawing

### CODE REVIEW

File: app/javascript/angular/code/services/graph.js [hunk 1]
```diff
@@ -3,176 +3,147 @@ import { buildOptions } from "./buildGraphOptions";
...
-              self.chart.redraw();
-              self.chart.hideLoading();
-            });
+let measurementsByTime = {};
+let chart = null;
+const RENDER_TO_ID = "graph";
+
+export const getInitialData = (selectedSensor, heat, singleFixedSession) => {
+  var end_date = new Date(singleFixedSession.endTime()).getTime();
+  var start_date = end_date - 24 * 60 * 60 * 1000;
```
Comment (gitjul) [inline app/javascript/angular/code/services/graph.js]: how about `let startDate` etc.?

File: app/javascript/angular/code/services/graph.js [hunk 2]
```diff
@@ -3,176 +3,147 @@ import { buildOptions } from "./buildGraphOptions";
...
+
+  var min1 = { count: 1, type: "minute", text: "1min" };
+  var min5 = { count: 5, type: "minute", text: "5min" };
+  var min30 = { count: 30, type: "minute", text: "30min" };
+  var hr1 = { count: 1, type: "hour", text: "1hr" };
+  var hrs12 = { count: 12, type: "hour", text: "12hrs" };
+  var hrs24 = { count: 24, type: "hour", text: "24hrs" };
+  var wk1 = { count: 1, type: "week", text: "1wk" };
+  var mth1 = { count: 1, type: "month", text: "1mth" };
+  var all = { type: "all", text: "All" };
```
Comment (gitjul) [inline app/javascript/angular/code/services/graph.js]: how about `let`?

File: app/javascript/angular/code/services/graph.js [hunk 3]
```diff
@@ -3,176 +3,147 @@ import { buildOptions } from "./buildGraphOptions";
...
+
+  const buttons = isFixed
+    ? [hr1, hrs12, hrs24, wk1, mth1, all]
+    : [min1, min5, min30, hr1, hrs12, all];
+  const selectedButton = isFixed ? 2 : 4;
```
Comment (gitjul) [inline app/javascript/angular/code/services/graph.js]: what do 2 and 4 represent? would it be a good idea to put them in constants?

File: app/javascript/angular/code/services/graph.js [hunk 4]
```diff
@@ -3,176 +3,147 @@ import { buildOptions } from "./buildGraphOptions";
...
+};
+
+const onMouseOverSingle = point => graphHighlight.show([point]);
+
+const onMouseOverMultiple = (start, end) => {
+  var startSec = start;
+  var endSec = end;
+  var points = [];
+  var point;
+  for (var i = startSec; i <= endSec; i = i + 1000) {
```
Comment (gitjul) [inline app/javascript/angular/code/services/graph.js]: `let`s?

File: app/javascript/angular/code/services/graph.js [hunk 5]
```diff
@@ -3,176 +3,147 @@ import { buildOptions } from "./buildGraphOptions";
...
+    point = measurementsByTime[i + ""];
+    if (point) points.push(point);
   }
-]);
+  var pointNum = Math.floor(points.length / 2);
```
Comment (gitjul) [inline app/javascript/angular/code/services/graph.js]: same here

File: app/javascript/angular/code/services/graph.js [hunk 6]
```diff
@@ -3,176 +3,147 @@ import { buildOptions } from "./buildGraphOptions";
...
+  graphHighlight.show([points[pointNum]]);
+};
+
+const afterSetExtremes = singleFixedSession => e => {
+  var final_point = {};
+  var end_time = new Date(singleFixedSession.endTime()).getTime();
```
Comment (gitjul) [inline app/javascript/angular/code/services/graph.js]: same here

**Comment by 3v0k4** [comment]
Thanks for the comments. I'm gonna take care of moving to ES6 (eg `let`s) and code style while I keep working on stuff

**Pull request merged**